Mathlib/Data/Matrix/Basis.lean

```diff
@@ -255,8 +257,20 @@ theorem mem_range_scalar_of_commute_stdBasisMatrix {M : Matrix n n α}
     · rw [col_eq_zero_of_commute_stdBasisMatrix (hM hkl.symm) hkl]
     · rw [row_eq_zero_of_commute_stdBasisMatrix (hM hij) hkl.symm]
 
-end Commute
+theorem mem_range_scalar_iff_commute_stdBasisMatrix {M : Matrix n n α} :
+    M ∈ Set.range (Matrix.scalar n) ↔ ∀ (i j : n), i ≠ j → Commute (stdBasisMatrix i j 1) M := by
+  refine ⟨fun ⟨r, hr⟩ i j _ => hr ▸ Commute.symm ?_, mem_range_scalar_of_commute_stdBasisMatrix⟩
+  rw [scalar_commute_iff, smul_stdBasisMatrix, smul_stdBasisMatrix, smul_eq_mul, op_smul_eq_mul,
+    mul_one, one_mul]
```

Comment:
```suggestion
  rewrite [scalar_commute_iff]
  simp
```
I find it convenient to have `simp` simplify it.

Reply:
Note that mathlib prefers `rw` over `rewrite`

Reply:
Because `rw [scalar_commute_iff]` didn't close the goal, I used `rewrite` instead.